### DISCUSSION TITLE
FreeBSD-provided packages: status

### DISCUSSION THREAD

**Discussion by grahamperrin** [helloSystem/hello#382] [Q&A] [answered]
<https://github.com/helloSystem/ISO/issues/141#issuecomment-1250129155> @probonopd asked: 

> 
![image](https://user-images.githubusercontent.com/2480569/190874780-e5ba66b6-dc82-4420-94c9-cc508011e799.png)
> How did you find that page, and what does it tell us?

**Comment by probonopd** [comment]
Somehow 
http://beefy14.nyi.freebsd.org/jail.html?mastername=130amd64-quarterly
seems to have that kind of information.

But how do I get the info why `print/hplip` is not built for `FreeBSD:14:amd64`?

`140amd64-quarterly` does not appear to exist, at least not on http://beefy14.nyi.freebsd.org... maybe on another server? https://pkg-status.freebsd.org/ may hold this information, albeit in a very cryptic form?

The maintainer of freshports.org is wondering the same thing:
https://github.com/FreshPorts/freshports/issues/384#issuecomment-1250131900

**Reply by grahamperrin** [reply]
Please see my suggested answer.

**Reply by grahamperrin** [reply]
> … why `print/hplip` is not built for `FreeBSD:14:amd64`? …

The screenshot in [the question](https://github.com/helloSystem/hello/discussions/382#discussion-4400586) showed the _Reason_ in this case: 

>> net-snmp-5.9.1_1,1

There's a log. Orientation: 

![image](https://user-images.githubusercontent.com/192271/190875755-cf96e630-e8ac-4e6c-9fee-0b2636436b8e.png)

**Comment by grahamperrin** [comment, marked as answer]
<https://docs.freebsd.org/en/articles/committers-guide/#ports-qa-misc-correctly-building>: 

> **How do I know if my port is building correctly or not?**
> 
> The packages are built multiple times each week. If a port fails, the maintainer will receive an email from `pkg-fallout@FreeBSD.org`.
> 
> Reports for all the package builds (official, experimental, and non-regression) are aggregated at [pkg-status.freebsd.org](https://docs.freebsd.org/en/articles/committers-guide/pkg-status.FreeBSD.org).

## Retrospective

The **Package availability in repositories** issue [began](https://github.com/helloSystem/ISO/issues/141#issue-807887964) with a `-CURRENT` example. 

A `12.2⋯` example appeared later in 2021 at <https://github.com/helloSystem/hello/discussions/224#discussion-3637718>. 

## Steps

### `FreeBSD:13:amd64`

1. <https://pkg-status.freebsd.org/builds?type=package>
2. seek _131amd64_
3. aim for the required row
4. click the Poudriere Build icon. 

Orientation: 

![image](https://user-images.githubusercontent.com/192271/190874739-87828042-6627-47ad-b653-5578fdc9ff13.png)

### `FreeBSD:14:amd64`

1. <https://pkg-status.freebsd.org/?type=package&all=1> (view all)
2. seek _main-amd64_
3. use the _Started (UTC)_ column to sort, if required
4. aim for the required row
5. click the Poudriere Build icon. 

Orientation: 

![image](https://user-images.githubusercontent.com/192271/190874946-86d26516-b0b8-4ee9-8194-a6915c9a27cc.png)

The somewhat different steps are to work around FreeBSD bug [260152 – Listings via pkg-status.freebsd.org appear outdated for main-amd64, main-arm64 (FreeBSD:14:aarch64) …](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=260152).

**Reply by probonopd** [reply]
Thanks @grahamperrin - very helpful. Albeit a bit more complicated than I was hoping for.

**Reply by grahamperrin** [reply]
The more recent answer https://github.com/helloSystem/hello/discussions/382#discussioncomment-6215850 refers to more recent documentation, published from my `freebsd.org` home directory.

**Comment by grahamperrin** [comment]
Via <https://old.reddit.com/r/freebsd/comments/102lt8a/-/> (2023-01-23): 

* [Using package build records at pkg-status.freebsd.org](https://people.freebsd.org/~grahamperrin/pkg-status/)
